src/community/netcdf-ghrsst/src/test/java/org/geoserver/wcs/responses/GHRSSTWCSTest.java

```diff
@@ -0,0 +1,218 @@
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+/**
+ * Base support class for NetCDF wcs tests.
+ *
+ * @author Daniele Romagnoli, GeoSolutions
+ */
+public class GHRSSTWCSTest extends WCSKVPTestSupport {
+
+    public static QName SST = new QName(CiteTestData.WCS_URI, "sst", CiteTestData.WCS_PREFIX);
+
+    @Override
+    protected void setUpTestData(SystemTestData testData) throws Exception {
+        // no extra data needed
+    }
+
+    @Override
+    protected void onSetUp(SystemTestData testData) throws Exception {
+        super.onSetUp(testData);
+        Catalog cat = getCatalog();
+
+        // setup the NetCDF
+        testData.addRasterLayer(SST, "sst-orbit053.nc", null, null, this.getClass(), cat);
+
+        // build a view putting toghether all bands
+        CoverageStoreInfo storeInfo = cat.getCoverageStores().get(0);
+        final CoverageView coverageView = buildSstView();
+        final CatalogBuilder builder = new CatalogBuilder(cat);
+        builder.setStore(storeInfo);
+        final CoverageInfo coverageInfo = coverageView.createCoverageInfo(SST.getLocalPart(), storeInfo, builder);
+        cat.add(coverageInfo);
+        LayerInfo layer = builder.buildLayer(coverageInfo);
+        cat.add(layer);
+
+        // enable time
+        setupRasterDimension(getLayerId(SST), ResourceInfo.TIME, DimensionPresentation.LIST, null);
+
+        // setup GHRSST output mode
+        CoverageInfo coverage = cat.getCoverageByName("wcs:sst");
+        NetCDFLayerSettingsContainer settings = new NetCDFLayerSettingsContainer();
+        settings.setCopyAttributes(true);
+        settings.setCopyGlobalAttributes(true);
+        settings.getMetadata().put(GHRSSTEncoder.SETTINGS_KEY, Boolean.TRUE);
+        coverage.getMetadata().put(NetCDFSettingsContainer.NETCDFOUT_KEY, settings);
+        cat.save(coverage);
+    }
+
+    private CoverageView buildSstView() {
+        String[] bandNames = new String[]{"pixels_per_bin", "sea_surface_temperature", "sst_dtime",
+                "quality_level", "wind_speed", "wind_speed_dtime_from_sst", "sea_ice_fraction",
+                "sea_ice_fraction_dtime_from_sst", "sses_bias", "sses_standard_deviation"};
+
+        List<CoverageView.CoverageBand> coverageBands = new ArrayList<>();
+        for (int i = 0; i < bandNames.length; i++) {
+            String bandName = bandNames[i];
+            final CoverageView.CoverageBand band = new CoverageView.CoverageBand(
+                    Arrays.asList(new CoverageView.InputCoverageBand(bandName, "0")), bandName, i,
+                    CoverageView.CompositionType.BAND_SELECT);
+            coverageBands.add(band);
+        }
+        final CoverageView coverageView = new CoverageView(SST.getLocalPart(), coverageBands);
+        return coverageView;
+    }
+
+    /**
+     * Test NetCDF output from a NetCDF file with a rotated pole projection.
+     */
```

Comment:
I would change the comment for this test to reflect a GHRSST write instead.

Reply:
Yep